docs/deployment/getting_started_locally.md

```diff
@@ -136,11 +136,51 @@ Please check your console output for the concrete port-forwarding on your machin
+### Readiness and Leader Election Checks
 
-Thus, leader election, health and readiness checks for `gardener-admission-controller`, `gardener-apiserver`, `gardener-controller-manager`, `gardener-scheduler`,`gardenlet` and `operator` are disabled when debugging.
+Standard Kubernetes health checks deserve attention when debugging Gardener using Delve. When Gardener controllers encounter an error condition (such as a missing CRD), they may enter (or remain in) an unready state. Under normal deployment
+circumstances with readiness checks enabled, the controllers will be restarted by Kubernetes until the undesirable preconditions clear and the controller can fully start. But this will not happen if the readiness checks are disabled for
+debugging purposes.
 
-If you have similar problems with other components which are not deployed by skaffold, you could temporarily turn off the leader election and disable liveness and readiness probes there too.
+This means that when a goroutine of gardenlet (or any other gardener-core component you try to debug) is paused on a breakpoint, all the other goroutines (including those responding to health checks) are also paused. Kubernetes controllers
+are *not* paused though. For instance, when the whole gardenlet process is paused, it can not renew its lease and can not respond to the liveness and readiness probes. The Kubernetes control plane interprets that controller as failed and
+attempts to restart it. That's obviously disruptive to a debugging session with Delve!
+
+To alleviate some of this, Skaffold automatically increases `timeoutSeconds` of liveness and readiness probes to 600, but sometimes this is not enough to resolve an issue -- or is too long for other controllers which depend on responses.
+The new developer will need to be initially cautious in their interpretations of these states, but they become second nature over time.
+
+Thus, new developers are advised to set `ENABLE_HEALTH_CHECKS`, both at the command line with the various make environments and with Google Cloud Code IDE plugin. This will cause health checks to be deployed by Helm as they would be in
+production. Note this is **_not_** a good long term debugging configuration though.
+
+```bash
+make gardener-debug ENABLE_HEALTH_CHECKS=admission,controller,scheduler,gardenlet,operator
+```
+
+Once a developer discovers a health check disturbing their debugging workflow, they should remove the check from the `ENABLE_HEALTH_CHECKS` list. The available checks are currently from the set
+of `{admission,controller,scheduler, gardenlet, operator}`.
```

Comment:
In my opinion the `ENABLE_HEALTH_CHECKS` environment variable is not useful. Debugging with enabled health checks does not work, because all go-routines are stopped at break points and the pods will be killed quite fast.
In practice nobody would use the parameter and in my opinion it is not a good idea to overload the skaffold setup with "once in a lifetime switches" for demo purposes only.
Additionally, I don't want to advice new developers to test non-working configurations, because this just annoys people.

Reply:
Please see general comments.